cmake/global_variables.cmake

```diff
@@ -53,4 +55,53 @@ if(NOT EXISTS "${ORT_LIB_DIR}/${ONNXRUNTIME_LIB}")
 endif()
 if(NOT EXISTS "${ORT_HEADER_DIR}/onnxruntime_c_api.h")
   message(FATAL_ERROR "Expected the ONNX Runtime C API header to be found at \"${ORT_HEADER_DIR}/onnxruntime_c_api.h\". Actual: Not found.")
-endif()
\ No newline at end of file
+endif()
+
+
+# normalize the target platform to x64 or arm64. additional architectures can be added as needed.
+if (MSVC)
+  if (CMAKE_VS_PLATFORM_NAME)
+    # cross-platform generator
+    set(genai_target_platform ${CMAKE_VS_PLATFORM_NAME})
+  else()
+    set(genai_target_platform ${CMAKE_SYSTEM_PROCESSOR})
+  endif()
+
+  if (genai_target_platform STREQUAL "arm64")
+    # pass
+  elseif (genai_target_platform STREQUAL "x64" OR 
+          genai_target_platform STREQUAL "x86_64" OR 
+          genai_target_platform STREQUAL "AMD64" OR 
+          CMAKE_GENERATOR MATCHES "Win64")
+    set(genai_target_platform "x64")
+  else()
+    message(FATAL_ERROR "Unsupported architecture. CMAKE_SYSTEM_PROCESSOR: ${CMAKE_SYSTEM_PROCESSOR}")
+  endif()
+elseif(APPLE)
+  if (CMAKE_OSX_ARCHITECTURES STREQUAL "arm64")
```

Comment:
This seems to be unset when I try to build on a Mac. Is there a way of passing it through?